nuxeo-features/nuxeo-platform-notification/nuxeo-platform-notification-core/src/test/java/org/nuxeo/ecm/platform/ec/notification/TestEmailNotification.java

```diff
@@ -0,0 +1,182 @@
+
+    @Inject
+    protected CoreSession session;
+
+    @Inject
+    protected NotificationManager notificationManager;
+
+    @Inject
+    protected EventService eventService;
+
+    @Inject
+    protected TransactionalFeature transactionalFeature;
+
+    @Inject
+    protected SmtpMailServerFeature.MailsResult emailsResult;
+
+    protected DocumentModel domain;
+
+    @Before
+    public void before() {
+        domain = session.createDocumentModel("/", "domain", "Domain");
+        domain = session.createDocument(domain);
+    }
+
+    @Test
+    public void shouldReceiveNotificationMailWhenSubscribeToDocument() {
+        // create a File document
+        DocumentModel documentModel = createDocument(domain, DOCUMENT_NAME);
+
+        // subscribe to the dummy notification
+        addSubscription(documentModel);
+
+        // fire the dummy event
+        DocumentEventContext ctx = new DocumentEventContext(session, session.getPrincipal(), documentModel);
+        Event event = ctx.newEvent(DUMMY_EVENT_NAME);
+        eventService.fireEvent(event);
+        transactionalFeature.nextTransaction();
+
+        // check the received mail
+        checkMailContent(documentModel, event);
+    }
+
+    @Test
+    public void shouldReceiveNotificationMailWhenSubscribeToParentDocument() {
+        DocumentModel parentDocModel = createDocument(domain, DOCUMENT_NAME);
+
+        // subscribe to the dummy notification on the parent document
+        addSubscription(parentDocModel);
+
+        // create a simple hierarchy (domain <- DOCUMENT_NAME <- ANOTHER_DOCUMENT_NAME)
+        DocumentModel mainDocModel = createDocument(parentDocModel, ANOTHER_DOCUMENT_NAME);
+
+        // fire the dummy event on the child document
+        DocumentEventContext ctx = new DocumentEventContext(session, session.getPrincipal(), mainDocModel);
+        Event event = ctx.newEvent(DUMMY_EVENT_NAME);
+        eventService.fireEvent(event);
+        transactionalFeature.nextTransaction();
+
+        // check the received mail
+        checkMailContent(mainDocModel, event);
+    }
+
+    protected void checkMailContent(DocumentModel documentModel, Event event) {
+        assertEquals(1, emailsResult.getMails().size());
+        SmtpMailServerFeature.MailMessage mailMessage = emailsResult.getMails().get(0);
+        // check the subject
+        assertEquals(String.format("[Dummy]Notification on the document '%s'", documentModel.getTitle()),
+                mailMessage.getSubject());
+
+        // check the text content
+        String expectedMailContent = getExpectedMailContent(documentModel, event);
+        assertEquals(expectedMailContent, mailMessage.getContent());
+    }
+
+    protected DocumentModel createDocument(DocumentModel parent, String name) {
+        DocumentModel documentModel = session.createDocumentModel(parent.getPathAsString(), name, "File");
+        return session.createDocument(documentModel);
+    }
```

Comment:
I haven't been conquered by this kind of method. 

We're having them in more and more test classes, leading me to the question: why don't we want it in the sources for all usage?

They can change from one test to another, which doesn't help when reviewing code as we can't be sure that method does the thing we're used to/expect to. Furthermore, I think it's surprising to have code which doesn't use this method in the same test (we can see it in the `@Before`). Surprising because the method has the same name than the core session one and because it's not coherent with rest of test.

When it comes to use it, we often need to put a comment in front of, in the end we don't benefit from the saved line this method gives.
For instance, I would prefer the "old way" in the text block below:
```java
// create a simple hierarchy (domain <- DOCUMENT_NAME <- ANOTHER_DOCUMENT_NAME)
DocumentModel mainDocModel = createDocument(parentDocModel, ANOTHER_DOCUMENT_NAME);
```
```java
DocumentModel mainDocModel = session.createDocumentModel("/domain/anyFile", ANOTHER_DOCUMENT_NAME, "File");
mainDocModel = session.createDocument(parentDocModel);
```
I'm in favor of removing this kind of method and doing light and explicit things.

Note: the example above highlights the fact you're creating a `File` under another `File` which is permitted since the comment improvement but I'm not sure this is what we want to do here, we may want to do something closer to reality.